client/interfaces.go

```diff
@@ -82,6 +91,12 @@ func (client *Client) Interfaces() (interfaces Interfaces, err error) {
 	return
 }
 
+// InterfaceInfos returns all interfaces, their meta-data, plugs and slots.
+func (client *Client) InterfaceInfos() (infos map[string]InterfaceInfo, err error) {
```

Comment:
I did something like this (separate methods) but still allowed the "list" method to return both name and summary.